config/src/main/java/com/typesafe/config/impl/SimpleConfigOrigin.java

```diff
@@ -550,4 +550,18 @@ static SimpleConfigOrigin fromBase(SimpleConfigOrigin baseOrigin,
         Map<SerializedField, Object> fields = applyFieldsDelta(baseFields, delta);
         return fromFields(fields);
     }
+
+    @Override
+    public ConfigOrigin withComments(List<String> comments) {
+        return this.setComments(comments);
+    }
+
+    @Override
+    public ConfigOrigin withLineNumber(int lineNumber) {
+        if (this.originType == OriginType.GENERIC) {
+            //This type should not have a lineNumber.
+            throw new UnsupportedOperationException();
+        }
```

Comment:
I think it would be OK not to throw an exception here; if someone has a description and a line number, that's fine. And then we could again just rename setLineNumber to withLineNumber. If we do throw an exception here it should be ConfigException.BugOrBroken, but I think it's simplest to just not throw one.